sdk_contrib/async_processor/src/main/java/io/opentelemetry/sdk/contrib/trace/export/DisruptorAsyncSpanProcessor.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.contrib.trace.export;
+
+import com.google.common.base.Preconditions;
+import com.lmax.disruptor.SleepingWaitStrategy;
+import com.lmax.disruptor.WaitStrategy;
+import io.opentelemetry.sdk.trace.ReadableSpan;
+import io.opentelemetry.sdk.trace.SpanProcessor;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * A {@link SpanProcessor} implementation that uses {@code Disruptor} to execute all the hooks on an
+ * async thread.
+ */
+@ThreadSafe
+public final class DisruptorAsyncSpanProcessor implements SpanProcessor {
+  // Number of events that can be enqueued at any one time. If more than this are enqueued,
+  // then subsequent attempts to enqueue new entries will block.
+  private static final int DEFAULT_DISRUPTOR_BUFFER_SIZE = 8192;
+  // The default value of the Disruptor behavior, blocks when no space available.
+  private static final boolean DEFAULT_BLOCKING = true;
```

Comment:
Per https://github.com/open-telemetry/opentelemetry-specification/pull/130 I think we want to make default non-blocking?

Reply:
Now that open-telemetry/opentelemetry-specification#130 is merged this default value should be false (can be done in another PR).